pkg/lib/authn/otp/code_store_redis.go

```diff
@@ -19,15 +19,15 @@ type CodeStoreRedis struct {
 	Clock clock.Clock
 }
 
-func (s *CodeStoreRedis) set(target string, code *Code) error {
+func (s *CodeStoreRedis) set(purpose string, target string, code *Code) error {
```

Comment:
Binding the OTP to purpose would make the generated OTP code via Admin API only usable for login, not verification.

Reply:
The admin API is `generateOOBOTPCode`, which (I think) implies the OTP code is usable for OOB-OTP login.

I'm concerned that the OTP code for different purposes could be mixed up, since the code valid period for different purpose could be different.

How about adding an optional field `purpose` for the mutation, with default value to login?